test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -421,6 +421,8 @@ TEST_F(testNuma, checkModeInterleaveCustomPartSize) {
     ASSERT_NO_FATAL_FAILURE(getNumaNodeByPtr(ptr, &index));
     for (size_t i = 0; i < (size_t)part_num; i++) {
         for (size_t j = 0; j < part_size; j += page_size) {
+            ASSERT_GE(index, 0);
+            ASSERT_LT(index, numa_nodes.size());
```

Comment:
This check seems to be wrong in some cases. Let's say that we have available `numa_nodes` vector looking like this: `0, 1, 4`. If memory is allocated on the node `#4`, then this check would fail the test, because the `numa_nodes` size is 3.

Reply:
But there is a line:
```c
index = (index + 1) % numa_nodes.size();
```
below inside this `for` loop, so it has to be true that: `index < numa_nodes.size();`

Reply:
Hmm, looks like an issue to be resolved in some other PR then... I think the test should cover cases that I mentioned. We can merge this one to satisfy Coverity, though.

Reply:
I do not get it. If `vector = [4]` then `index == 0 and vector[0] == 4` - am I wrong?

Reply:
`getNumaNodeByPtr` returns the actual node number where the memory has been allocated, so it will return in this case `index=4`.

Reply:
If  `index` in `numa_nodes[index]` can be greater than `numa_nodes.size()` there is a real bug in this test.
@lplewa could you fix that?

Reply:
Fixed.

Done